File: plugin/src/main/java/de/s42/tradecycle/strategy/ShiftInteractStrategy.java
package de.s42.tradecycle.strategy;

import de.s42.tradecycle.event.VillagerCycleTradeEvent;
import org.bukkit.Bukkit;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.entity.Villager;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEntityEvent;
import org.jetbrains.annotations.NotNull;

public class ShiftInteractStrategy implements Listener {

    @EventHandler
    public void onPlayerInteractEntityEvent(@NotNull PlayerInteractEntityEvent event) {
        if (event.isCancelled()) {
            return;
        }

        Entity entity = event.getRightClicked();
        if (!(entity instanceof Villager villager)) {
            return;
        }

        Player player = event.getPlayer();
        if (!player.isSneaking()) {
            return;
        }

        Bukkit.getPluginManager().callEvent(new VillagerCycleTradeEvent(player, villager));
    }
}

File: core/src/main/kotlin/tr/s42/core/strategy/ShiftInteractStrategy.kt
package net.cutecraft.core.strategy

import net.cutecraft.core.event.VillagerCycleTradeEvent
import org.bukkit.Bukkit
import org.bukkit.entity.Villager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEntityEvent

class ShiftInteractStrategy : Listener {

    @EventHandler
    fun onPlayerInteractEntityEvent(event: PlayerInteractEntityEvent) {
        if (event.isCancelled) {
            return
        }

        val entity = event.rightClicked
        if (entity !is Villager) {
            return
        }

        val player = event.player
        if (!player.isSneaking) {
            return
        }

        Bukkit.getPluginManager().callEvent(VillagerCycleTradeEvent(player, entity))
    }

}
